src/field_theory/ratfunc.lean

```diff
@@ -1095,6 +1095,40 @@ begin
   { exact algebra_map_ne_zero (denom_ne_zero y) },
 end
 
+lemma nat_degree_well_defined {r₁ r₂ s₁ s₂ : polynomial K} (hr₁ : r₁ ≠ 0) (hs₁ : s₁ ≠ 0)
+  (hr₂ : r₂ ≠ 0) (hs₂ : s₂ ≠ 0) (h_eq : r₁*s₂ = r₂*s₁) :
+  (r₁.nat_degree : ℤ) - s₁.nat_degree = (r₂.nat_degree : ℤ) - s₂.nat_degree :=
+begin
+  rw sub_eq_sub_iff_add_eq_add,
+  norm_cast,
+  rw [← nat_degree_mul hr₁ hs₂, ← nat_degree_mul hr₂ hs₁, h_eq]
+end
+
+lemma num_add_ne_zero {x y : ratfunc K} (hxy : x + y ≠ 0) :
```

Comment:
The name of this lemma and the two lemmas below it don't really follow the convention.
How about `num_mul_denom_add_denom_mul_num_ne_zero`?

Reply:
I've renamed the three lemmas. I'm not sure about the two `nat_degree` lemmas, I think that if I try to follow convention too closely the names get too long.